sdk/src/androidTest/java/com/mapbox/maps/CameraAnimationsPluginTest.kt

```diff
@@ -451,6 +460,123 @@ class CameraAnimationsPluginTest : BaseAnimationMapTest() {
     }
   }
 
+  @Test
+  fun testEaseToStartDelayCanceled() {
+    if (ignoreTestForGivenAbi()) {
+      return
+    }
+    val targetBearing1 = 5.0
+    val targetBearing2 = 10.0
+    val targetBearing3 = 15.0
+    val cameraOptions1 = CameraOptions.Builder().bearing(targetBearing1).build()
+    val cameraOptions2 = CameraOptions.Builder().bearing(targetBearing2).build()
+    val cameraOptions3 = CameraOptions.Builder().bearing(targetBearing3).build()
+    val expectedValues = mutableSetOf(targetBearing1, targetBearing3)
+    val updatedValues = mutableListOf<Double>()
+
+    val cameraAnimationPlugin = mapView.camera
+    val latch = CountDownLatch(2)
+    cameraAnimationPlugin.addCameraBearingChangeListener {
+      Logger.i(TAG, "onChanged $it")
+      updatedValues.add(it)
+      latch.countDown()
+    }
+
+    mainHandler.post {
+      cameraAnimationPlugin.easeTo(
+        cameraOptions1,
+        mapAnimationOptions {
+          duration(1)
+          startDelay(1)
+        }
+      )
+    }
+
+    mainHandler.postDelayed(
+      {
+        cameraAnimationPlugin.easeTo(
+          cameraOptions2,
+          mapAnimationOptions {
+            duration(1)
+            startDelay(3000)
+          }
+        )
+      },
+      100
+    )
+
+    // cameraOptions2 will be canceled since it's still waiting to start
+    mainHandler.postDelayed(
+      {
+        cameraAnimationPlugin.easeTo(
+          cameraOptions3,
+          mapAnimationOptions {
+            duration(1)
+            startDelay(0)
+          }
+        )
+      },
+      500
+    )
+
+    if (latch.await(LATCH_MAX_TIME, TimeUnit.MILLISECONDS)) {
+      assertArrayEquals(expectedValues.toDoubleArray(), updatedValues.toDoubleArray(), EPS)
+    } else {
+      throw TimeoutException()
+    }
```

Comment:
that's not reliable way to verify
some animation way still be running / scheduled although it should not while we will count down and report test as 🟢 .

Let's decrease delay values (use let's say 1000 instead of 3000), remove `latch.countDown()` at all and verify result like that (without if-else):
```
// wait for a bit more time than all delays in test
latch.await(1200, TimeUnit.MILLISECONDS)
// verify that data came as expected
assertArrayEquals(expectedValues.toDoubleArray(), updatedValues.toDoubleArray(), EPS)
```

Reply:
Aren't the rest of tests here the same?..

Reply:
they are not testing delays as those tests introduced - we basically want here to wait till time when everything delayed is executed (or canceled)